src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -424,6 +420,32 @@ private function getShell()
     }
 
     /**
+     * Reads user input.
+     *
+     * @param resource $stream The input stream
+     *
+     * @return string User input
+     *
+     * @throws \RuntimeException
+     */
+    private function readFromInput($stream)
+    {
+        if (STDIN === $stream && function_exists('readline')) {
+            $ret = readline();
+        } else {
+            $ret = fgets($stream, 4096);
+
+            if (false === $ret) {
+                throw new \RuntimeException('Aborted');
+            }
+
+            $ret = trim($ret);
```

Comment:
Thanks for your review.
I didn't know about `stream_get_line()` method. Would there be an advantage of using it on platforms not supporting `readline()`? The string would need to be trimmed as well if we want to keep the same behavior (removing whitespaces around the string) and arrow keys can't be used as well, so it doesn't seem there is any benefit.

I agree with you on the second point I should trim the string both with `readline` and with the usual `fgets` functions.